packages/suite/src/components/suite/Preloader/index.tsx

```diff
@@ -3,78 +3,40 @@ import React, { useEffect } from 'react';
 import { SUITE } from '@suite-actions/constants';
 import { SuiteLayout } from '@suite-components';
 import InitialLoading from './components/InitialLoading';
-import DiscoveryLoader from '@suite-components/DiscoveryLoader';
-import { Modals, RawModals } from '@suite-components/modals';
-import * as routerActions from '@suite-actions/routerActions';
 import DatabaseUpgradeModal from './components/DatabaseUpgradeModal';
 import PrerequisiteScreen from './components/PrerequisiteScreen';
 import { useDiscovery, useSelector, useActions } from '@suite-hooks';
-import Firmware from '@firmware-views';
-import FirmwareCustom from '@firmware-views/FirmwareCustom';
-import Recovery from '@recovery-views';
-import Backup from '@backup-views';
 import Onboarding from '@onboarding-views';
 import { getPrerequisites } from '@suite-utils/prerequisites';
 import ErrorPage from '@suite-views/error';
-import { Bridge, Udev, SwitchDevice, Version } from '@suite-views';
 import { useGuideKeyboard } from '@guide-hooks';
 
 import type { AppState } from '@suite-types';
 
-const getForegroundApplication = (route: AppState['router']['route']) => {
-    if (!route) return;
-    switch (route.app) {
-        case 'onboarding':
-            return Onboarding;
-        case 'firmware':
-            return Firmware;
-        case 'firmware-custom':
-            return FirmwareCustom;
-        case 'bridge':
-            return Bridge;
-        case 'udev':
-            return Udev;
-        case 'version':
-            return Version;
-        case 'switch-device':
-            return SwitchDevice;
-        case 'recovery':
-            return Recovery;
-        case 'backup':
-            return Backup;
-        default:
-            break;
-    }
-};
+const getFullscreenApp = (route: AppState['router']['route']) =>
```

Comment:
Maybe make it a switch? Since it's implied there could be more than 1 fullscreen app

Reply:
I hope there won't be :) But https://github.com/trezor/trezor-suite/pull/5313/commits/26c48125c1fec80d08b020092fccb6c0400ddbc1

Reply:
Yeah, it's just since there is already a function that selects an app to render and not a direct render a switch still makes more sense IMO c: